src/utils/twoFactor.js

```diff
@@ -186,11 +187,22 @@ export class TwoFactor {
         const contractBytes = new Uint8Array(await (await fetch('/multisig.wasm')).arrayBuffer())
         const { accountId } = accountData
         const account = this.wallet.getAccount(accountId)
-        const accountKeys = await account.getAccessKeys();
+        // replace account keys & recovery keys with limited access keys; DO NOT replace seed phrase keys
+        const accountKeys = (await account.getAccessKeys()).map((ak) => ak.public_key)
+            .filter((k) => k.indexOf('9t9') === -1)
         const recoveryMethods = await this.wallet.getRecoveryMethods()
-        const recoveryKeysED = recoveryMethods.data.map((rm) => rm.publicKey)
-        const fak2lak = recoveryMethods.data.filter(({ kind, publicKey }) => kind !== 'phrase' && publicKey !== null).map((rm) => toPK(rm.publicKey))
-        fak2lak.push(...accountKeys.filter((ak) => !recoveryKeysED.includes(ak.public_key)).map((ak) => toPK(ak.public_key)))
+        const seedPhraseKeys = recoveryMethods.data
+            .filter(({ kind, publicKey }) => kind === 'phrase' && publicKey !== null)
+            .map((rm) => rm.publicKey)
+            .filter((k) => !!k && accountKeys.includes(k))
+        const fak2lak = [...new Set([
+            ...accountKeys.filter((k) => !seedPhraseKeys.includes(k)),
+            ...recoveryMethods.data
```

Comment:
I don’t understand this, why we need to include anything not in the accountKeys?

Reply:
include all account keys that are not seed phrase keys, we leave those as FAKs

Reply:
@mattlockyer i understand that we need to include accountKeys. I don’t understand why ‘recoveryMethods’ part is needed. Like what extra keys it will add and why?

Reply:
ahhh I see. Makes sense. I will remove and retest with this.